### ISSUE TITLE
grunt-sass npm module (0.17.0) install fails

### ISSUE THREAD

**Issue by Masahigo** [Sitefinity/feather#1754]
Hi,

I just tried to run the "npm install" command from inside the Bootstrap subfolder under ResourcePackages following the instructions on the wiki (https://github.com/Sitefinity/feather/wiki/Use-a-framework-package). I have a fresh install of latest Sitefinity (8.0.5713.0) and Feather (1.2.210.0) installed and use latest version of NodeJS + npm.

It seems there is a problem with this particular npm module (grunt-sass) and its specific version (0.17.0, https://github.com/Sitefinity/feather-packages/blob/master/Bootstrap/package.json).

Error message (from PowerShell in admin mode)
************************************************************************
`win32-x64` exists; testing
module.js:355
  Module._extensions[extension](this, filename);
                               ^
Error: The specified procedure could not be found.
C:\Sitefinity\Testing Feather\Telerik.Sitefinity.Samples.EmptyProject\SitefinityWebApp\ResourcePackages\BootstrapCustom\
node_modules\grunt-sass\node_modules\node-sass\vendor\win32-x64\binding.node
    at Error (native)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (C:\Sitefinity\Testing Feather\Telerik.Sitefinity.Samples.EmptyProject\SitefinityWebApp\Resour
cePackages\BootstrapCustom\node_modules\grunt-sass\node_modules\node-sass\lib\index.js:181:15)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "i
nstall" "grunt-sass@0.17.0"
npm ERR! node v0.12.3
npm ERR! npm  v2.9.1
npm ERR! code ELIFECYCLE
************************************************************************

This same error can be reproduced by trying to install it seperately 
$ npm install grunt-sass@0.17.0

Newer version of grunt-sass install works
$ npm install --save-dev grunt-sass

After installing the newest version of grunt-sass (https://github.com/sindresorhus/grunt-sass, 1.0.0) and running the "grunt" I get the following message
----------------------------------------------------------

PS C:\Sitefinity\feather45\Website\ResourcePackages\BootstrapCustom> grunt
Running "newer:sprite" (newer) task

Running "newer:sprite:all" (newer) task
No newer files to process.

Running "sass:dist" (sass) task

Running "cssmin:minify" (cssmin) task
File assets/dist/css/styles.min.css created: 169.12 kB → 134.69 kB

Running "uglify:dist" (uglify) task
[TypeError: Cannot assign to read only property 'warnings' of true]
>> Uglifying source node_modules/bootstrap-sass/assets/javascripts/bootstrap.js failed.
Warning: Uglification failed.
Cannot assign to read only property 'warnings' of true.
 Use --force to continue.

Aborted due to warnings.
-----------------------------------------------------------------------------------------------

So basically now I run into trouble with bootstrap-sass module which is older in comparison.

Can you advise?

**Comment by ElenaGaneva** [comment]
Hi Masahigo,

I have been looking into the stack trace that you provide and I believe that there is an issue with grunt-sass version 0.17.0 and the versions of npm and node that you have. Could you please try to downgrade npm to 1.4.X and run npm-install and grunt again.

Furthermore I will add this to our backlog so we can update versions of dependent modules in package.js to latest versions.

Regards,
Elena

**Comment by Masahigo** [comment]
Thank you, I was able to get it working following these steps

- Download npm 1.4.3 (http://nodejs.org/dist/v0.10.26/node-v0.10.26-x86.msi) and run the installer
* this also includes older version of Node.js (0.10.26)
- Start the Node.js command prompt (Start > Node.js)
- Navigate to the ResourcePackages subfolder and run the commands
1) $ npm install
* Installed everything else normally except for the grunt-sass npm module
2) $ npm install grunt-sass@0.17.0
* This time the install succeeds (installs also node-sas@1.2.3)
3) $ grunt
* This time everything works as designed and the Grunt task is left running in console
